### PR TITLE
Dynamic forms task data context

### PR DESCRIPTION
**Pull request by sergei-maertens**
Alternative implementation for #178 

- [x] Working POC
- [x] Tests

### CODE REVIEW

File: backend/src/zac/camunda/user_tasks/context.py [hunk 1]
```diff
@@ -34,7 +34,14 @@ def get_context(task: Task) -> Optional[Context]:
...
     callable.
     """
-    (callback, *rest) = REGISTRY.get(task.form_key)
+    lookup = task.form_key
+    if task.form_key not in REGISTRY:
+        warnings.warn(
+            f"Unknown task registry key: '{lookup}'. Falling back to dynamic form.",
+            RuntimeWarning,
+        )
+        lookup = ""
```
Comment (damm89) [inline backend/src/zac/camunda/user_tasks/context.py:43]: What's your reasoning for not making lookup explicitly "zac:dynamicForm" or whatever? Wouldn't it be clearer for the API docs if it were to be explicit?

File: backend/src/zac/camunda/dynamic_forms/serializers.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        "int": IntSerializer,
+        "boolean": BooleanSerializer,
+        "date": DatetimeSerializer,
+    }
+
+    name = serializers.CharField(
+        label=_("Field name/identifier"),
+        required=True,
+    )
+    title = serializers.CharField(
```
Comment (damm89) [inline backend/src/zac/camunda/dynamic_forms/serializers.py:81]: Kelvin and I had agreed on label instead of title, but I guess he can change it back.

**Review by damm89 — APPROVED** [review]
Prima!

File: backend/src/zac/camunda/user_tasks/context.py (hunk 1 above)
Comment (sergei-maertens) [reply backend/src/zac/camunda/user_tasks/context.py:43]: I'd like to mimic what's the value in Camunda - which is empty normally for these cases.

Figuring out why the value is `zac:dynamicForm` to debug what's happening when you can't find that string anywhere in the process model BPMN would probably be confusing.

File: backend/src/zac/camunda/dynamic_forms/serializers.py (hunk 2 above)
Comment (sergei-maertens) [reply backend/src/zac/camunda/dynamic_forms/serializers.py:81]: I'll change it, I prefer label too

**Pull request merged**